File: ts/test-util.ts
import fs from "fs";

export const BOLD = "\u001b[1m";
export const GRAY = "\u001b[38;5;248m";
export const CYAN = "\u001b[36m";
export const RED = "\u001b[31m";
export const GREEN = "\u001b[32m";
export const YELLOW = "\u001b[33m";
export const RESET = "\u001b[0m";

export function fail(message: string) {
  console.log(RED + BOLD + "FAIL: " + RESET + message);
  process.exit(1);
}

export function check(condition: any, message: string): asserts condition {
  if (!condition) {
    fail(message);
  }
}

export function readLines(path: string): string[] {
  return fs
    .readFileSync(path, "utf-8")
    .split("\n")
    .map((l) => l.trim())
    .filter(Boolean);
}

File: csharp/TestUtil.cs
public static class TestUtil
{

    public const string BOLD = "\u001b[1m";
    public const string GRAY = "\u001b[38;5;248m";
    public const string CYAN = "\u001b[36m";
    public const string RED = "\u001b[31m";
    public const string GREEN = "\u001b[32m";
    public const string YELLOW = "\u001b[33m";
    public const string RESET = "\u001b[0m";

    public static void Check(bool condition, string message)
    {
        if (!condition)
        {
            Fail(message);
        }
    }

    public static void Fail(string message)
    {
        Console.WriteLine($"{RED}FAIL:{RESET} {message}");
        Environment.Exit(1);
    }
}